sys/net/routing/aodvv2/aodv.c

```diff
@@ -108,6 +120,63 @@ void aodv_set_metric_type(aodvv2_metric_t metric_type)
     }
     _metric_type = metric_type;
 }
+/*
+ * @brief handles callbacks from the FIB when it needs a route
+ */
+void *fib_signal_handler_thread(void *arg)
+{
+    (void) arg;
+    ipv6_addr_t dest;
+    struct netaddr na_dest;
+
+    fib_register_rp((uint8_t*) &aodvv2_prefix, aodvv2_prefix_len);
+
+    while (true) {
+        AODV_DEBUG("%s()\n", __func__);
+        msg_t msg;
+        msg_receive(&msg);
+
+        if (msg.type == FIB_MSG_RP_SIGNAL) {
+            rp_address_msg_t* rp_msg = (rp_address_msg_t*)msg.content.ptr;
+            if (rp_msg->address_size == sizeof(ipv6_addr_t)) {
+                /* We currently only support IPv6*/
+                memcpy(&dest, rp_msg->address, rp_msg->address_size);
+                /* Reply to the FIB so that it can stop blocking */
+                msg_reply(&msg, &msg);
+
+                /* perform/initiate a rreq for dst here*/
+                ipv6_addr_t_to_netaddr(&dest, &na_dest);
+
+                aodvv2_seqnum_t seqnum = seqnum_get();
+                seqnum_inc();
+
+                /* Build new RREQ */
+                struct aodvv2_packet_data rreq_data = (struct aodvv2_packet_data) {
+                    .hoplimit = AODVV2_MAX_HOPCOUNT,
+                    .metricType = _metric_type,
+                    .origNode = (struct node_data) {
+                        .addr = na_local,
+                        .metric = 0,
+                        .seqnum = seqnum,
+                    },
+                    .targNode = (struct node_data) {
+                        .addr = na_dest,
+                    },
+                    .timestamp = (timex_t) {0,0} /* this timestamp is never used, it exists
+                                                  * merely to make the compiler shut up */
+                };
+
+                DEBUG("\tstarting route discovery towards %s... \n",
+                      ipv6_addr_to_str(addr_str, IPV6_MAX_ADDR_STR_LEN, &dest));
+                aodv_send_rreq(&rreq_data);
+            }
+            else {
+                /* Reply to the FIB so that it can stop blocking */
+                msg_reply(&msg, &msg);
+            }
+        }
+    }
```

Comment:
please add `return NULL;` before the final closing brackets to match the requested return value